src/services/TelemetryAPI.js

```diff
@@ -0,0 +1,57 @@
+import axios from "axios";
+
+const TELEMETRY_API_URL = process.env.REACT_APP_TELEMETRY_API_URL;
+
+class TelemetryAPI {
+    static async getHeartRateAndOxygenSaturation(videoUrl, token) {
+        let heartRateAndOxygenSaturation = {"heartRate": "N/A", "oxygenSaturation": "N/A"};
+        if (videoUrl !== '') {
+            try {
+                const response = await axios({
+                    url: TELEMETRY_API_URL + '/telemetry-reports',
+                    params: {
+                        "telemetry_source_id": videoUrl,
+                    },
+                    headers: {
+                        'Authorization': `Bearer ${token}`
+                    },
+                    method: 'get'
+                });
+                const telemetryResult = response.data;
+                heartRateAndOxygenSaturation.heartRate =
+                    telemetryResult.telemetry_report.bpm.toFixed(2) + " ppm";
+                heartRateAndOxygenSaturation.oxygenSaturation =
+                    telemetryResult.telemetry_report.spo2.toFixed(2) + " %";
+            } catch (error) {
+                console.error(error);
+            }
+        }
+        return heartRateAndOxygenSaturation;
+    }
+
+    static async getBreathingFrequency(audioUrl, token) {
+        let breathingFrequency = {"breathingFrequency": "N/A"};
+        if (audioUrl !== '' && audioUrl !== undefined) {
```

Comment:
```suggestion
        if (audioUrl) {
```

Reply:
The thing is that the request should be triggered only if the audioUrl is different from ''. At the moment, the audioUrl is undefined.

Reply:
But as `''` is falsy, this will cover your case. Anyway, it's just a small detail.

Reply:
Ok, I didn't know that. Let me do this change and thanks for the suggestion. It's cleaner.

Reply:
Great! 😄